### ISSUE TITLE
Error when using embroider: Could not find module @ember-data/model/-private

### ISSUE THREAD

**Issue by NullVoxPopuli** [emberjs/data#7154]
### Reproduction

Please provide one of the following:

- a PR linking to this issue with a failing test showing the issue
- a [Twiddle](https://ember-twiddle.com/) with a simplified reproduction and instructions for how to
  observe the unexpected result.
- a github repository with a simplified reproduction and instructions for how to observe the unexpected result

[emberclear](https://emberclear.io) PR: https://github.com/NullVoxPopuli/emberclear/pull/784
```
cd packages/frontend
yarn start
# open browser to https://localhost:4201
# notice the following error in the console:
```
![image](https://user-images.githubusercontent.com/199018/80593292-f0006180-89ee-11ea-9f65-479989fe06bb.png)
 
Looks like the error is happening during the initial application transition:

![image](https://user-images.githubusercontent.com/199018/80593958-1b378080-89f0-11ea-8215-9e6de5cfb2c3.png)

which kinda makes sense, because I'm interacting with ember-data in the application route's beforeModel hook. But also mostly surprised this error wasn't thrown earlier? before visiting the page?

cc @ef4

### Versions

```cli
yarn list v1.22.4
├─ @ember-data/canary-features@3.18.0
├─ @ember-data/debug@3.18.0
├─ @ember-data/model@3.18.0
├─ @ember-data/private-build-infra@3.18.0
├─ @ember-data/record-data@3.18.0
├─ @ember-data/rfc395-data@0.0.4
├─ @ember-data/store@3.18.0
├─ @types/ember-data__model@3.15.0
├─ @types/ember-data__store@3.15.1
├─ @types/ember-data@3.1.9
└─ babel-plugin-ember-data-packages-polyfill@0.1.2
```

**Comment by runspired** [comment]
@nullvoxpopuli embroider is bringing a customization for EmberData that needs removed. EmberData is (and has been for a while) incompatible with that customization. While the error is recent your code would not have been built correctly in the past.

**Comment by NullVoxPopuli** [comment]
what is that and how do I disable it / remove it?
why does the customization exist in the first place?

**Comment by NullVoxPopuli** [comment]
Looks like this comment solves it: https://github.com/embroider-build/embroider/issues/396#issuecomment-611885598

@ef4 -- why is this needed?

**Comment by rreckonerr** [comment]
Seeing the same issue on e-data v4.4 and embroider v1.6

**Comment by rreckonerr** [comment]
The issue is solved in e-data v4.7 🎉
